metadeploy/api/models.py

```diff
@@ -649,8 +660,18 @@ class Job(HashIdMixin, models.Model):
         ClickThroughAgreement, on_delete=models.PROTECT, null=True
     )
 
-    def subscribable_by(self, user):
-        return self.is_public or user.is_staff or user == self.user
+    def subscribable_by(self, user, session):
```

Comment:
A docstring explaining the cases this is handling would be appreciated by our future selves, I'm sure.